rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromReturnNewRector/Fixture/skip_possible_void_with_early_return.php.inc

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromReturnNewRector\Fixture;
+
+final class SkipPossibleVoidWithEarlyReturn
+{
+    public function bar()
+    {
+        if (random_int(0,1)) {
+            return;
+        }
+
+        return new self();
```

Comment:
possible void without root return fixture seems needed, eg:

```php
function foo()
{
    if (rand(0,1)) {
        return new stdClass();
   }
}
```

I think `ReturnTypeInferer` service is more mature way to detect that, also allow verify code inside `while (true)`, `do { } while`, if else deep etc

Reply:
What do you suggest?

Reply:
The existing code:

```php
$returnType = $this->returnTypeInferer->inferFunctionLike($node);
if ($returnType instanceof UnionType) {
   return null;
}
```

as current verify `UnionType` is already fine for me.

Reply:
see existing code https://getrector.com/demo/e688fb7f-02ce-4e8b-8015-73877dd62a9a already skipped as expected.

Reply:
That's miss use of unrelated service that accidentally works here :)

This rule is only to add types for instance of `new` when possible. Union suggests PHP 8 is involved, which is not valid here.

Reply:
That means `SilentVoidResolver` should be used instead, that's more correct service to be used

Reply:
I think that's not neccesary either, as it checks inverted condition.

Reply:
ok 👍